org.librarysimplified.r2.vanilla/src/main/java/org/librarysimplified/r2/vanilla/internal/SR2Controller.kt

```diff
@@ -845,16 +832,19 @@ internal class SR2Controller private constructor(
       try {
         try {
           future.get()
+          this.longRunningCommands.remove(command)
           this.publishCommmandSucceeded(command)
         } catch (e: ExecutionException) {
           throw e.cause!!
         }
       } catch (e: SR2WebViewDisconnectedException) {
         this.logger.debug("webview disconnected: could not execute {}", command)
         this.eventSubject.onNext(SR2Event.SR2Error.SR2WebViewInaccessible("No web view is connected"))
+        this.longRunningCommands.remove(command)
         this.publishCommmandFailed(command, e)
       } catch (e: Exception) {
         this.logger.error("{}: ", command, e)
+        this.longRunningCommands.remove(command)
```

Comment:
Might be better to do this removal in a `finally` block.

Reply:
That's what I thought first, and then I was worried about the order of execution. I felt like the state should be changed before the command result event is published. Just in case subscribers rely on it. In practice event reception is delayed by the Android handler, so I suppose it doesn't matter.

Reply:
I agree, the event should be published only after the state is changed.

Reply:
Are you suggesting we can still do that with a `finally block`?

Reply:
Not without making the code hard to read.

Reply:
Ok, we agree on that.